### PR TITLE
Feature/meetings index page template

### PR DESCRIPTION
**Pull request by johnnyporkchops**
New meetings index page based on https://github.com/18F/fec-cms/issues/1138

I think this is mostly complete but there are some issues with the pagination. @xtine and @noahmanger 
As far as I can tell, I need separate pagination for each tab section and the logic is conflicting between tabs. I need to get some pairing help on the pagination functionality or research it more myself

Also this question from the issue still need to be resolved:
Question: @noahmanger do you think we should we add filter validation like in the data tables to show that results have changed/search term applied?

I still have some text changes to make and design tweaks but I wanted to get the functionality up for review sooner than later

### CODE REVIEW

File: fec/home/templates/home/commission_meetings.html [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        <label for="search" class="label">Search</label>
+        <input id="search" class="combo__input" name="search" type="text" value="{{ search }}">
+        <button type="submit" class="combo__button button button--standard button--search"><span class="u-visually-hidden">Search</span></button>
+      </div>
+    </div>
+</form>
+{% endblock %}
+
+{% block feed %}
+<section id="homepage-help-candidate-and-committees" class="slab tab-interface">
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Need to rename this id to reflect meetings page.

File: fec/home/templates/home/commission_meetings.html [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  <div class="container">
+    <div class="slab slab--home grid">
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Remove this slab.

File: fec/home/templates/home/commission_meetings.html [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+
+
+
+{% endblock %}
+
+
+
+{% block extra_js %}
+  <script type="text/javascript" src="{% static 'js/home.js' %}"></script>
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Remove the `extra_js` block.

File: fec/home/templates/home/commission_meetings.html [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+                 <p>FEC.gov is in the middle of a redesign, and some pages aren’t included in search results yet. Please try another search. Or, <a href="{{ settings.FEC_CLASSIC_URL }}">search the archive</a> of FEC.gov’s previous design.</p>
+                 <p>If you’d like to contact our team, we’re available by <a href="mailto:{{ settings.WEBMANAGER_EMAIL }}">email</a> or <a href="https://github.com/18f/fec">GitHub</a>.</p>
+               </div>
+            </div>
+              {% endif %}
+          </section>
+    </div>
+  </div>
+</section>
+
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Try to tighten up the extra spacings.

File: fec/fec/templates/partials/meeting.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+
+{% load wagtailimages_tags %}
```
Comment (xtine) [inline fec/fec/templates/partials/meeting.html]: Don't need wagtail images here.

File: fec/fec/templates/partials/meeting.html [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+{% load filters %}
+        <tr class="simple-table__row">
+        <td class="simple-table__cell">{{ meeting.date|date:'F j, Y' }}</td>
+        {% if meeting.sunshine_act_links %}
+          {% for link in meeting.sunshine_act_links|splitlines %}
+          <td class="simple-table__cell"> <a class="pdf-link" href="{{ link }}">{% if forloop.counter >= 2 %}Amended{% endif %}
```
Comment (xtine) [inline fec/fec/templates/partials/meeting.html]: Remove `class="pdf-link"`.

File: fec/home/templates/home/commission_meetings.html [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+            {% if executive_sessions %}
+               <table class="simple-table simple-table">
+                <thead class="simple-table__header">
+                  <tr>
+                    <th class="simple-table__header-cell">Meeting Date</th>
+                    <th class="simple-table__header-cell">Sunshine Act notices<span class="simple-table__header-subtext"></span></th>
+                  </tr>
+                </thead>
+                  <tbody>
+                    {% for executive_session in executive_sessions reversed %}
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Put the for loop within `partials/meeting.html` template.

File: fec/fec/templates/partials/meeting.html (hunk 6 above)
Comment (xtine) [inline fec/fec/templates/partials/meeting.html]: These do not need to be within `<td>` tags, they can just be appended as regular links, in list form.

File: fec/home/templates/home/commission_meetings.html [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+          </nav>
+        </div>
+
+        <div class="main__content--right-full">
+
+          <section id="meetings-meetings-section-1" class="option" role="tabpanel" aria-hidden="true" aria-labelledby="open-meetings">
+            <h2 id="meetings-meetings-section-1-heading">Open Meetings</h2>
+             <p>Open to the public. The Commission considers new regulations, advisory opinions and other public matters at open meetings, which are typically held at FEC headquarters on Thursdays at 10:00am. Commission meeting agendas are usually published a week before a scheduled meeting.</p>
+            {% if open_meetings %}
+               <table class="simple-table simple-table">
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Can remove the extra `simple-table` class declaration.

File: fec/home/templates/home/commission_meetings.html (hunk 3 above)
Comment (johnnyporkchops) [inline fec/home/templates/home/commission_meetings.html]: OK, but the tabs did not work without it (?)

File: fec/home/templates/home/commission_meetings.html (hunk 7 above)
Comment (johnnyporkchops) [inline fec/home/templates/home/commission_meetings.html]: There are three different for loops, putting then inside the partial  would mean making three partials, one for each. Having it outside the partial is how it is on home/templates/home/latest_updates.html
```{% for update in updates %}
      {% include 'partials/update.html' with update=update show_tag=True %}
    {% endfor %}
 ```

File: fec/home/templates/home/commission_meetings.html (hunk 3 above)
Comment (johnnyporkchops) [inline fec/home/templates/home/commission_meetings.html]: @xtine wrote in Slack: oh shoot. yeah you need the js, but you don’t need the homepage specific one. let me look up the generic one


John Carroll [6:38 PM] 
cool


Christine Cheung (18F - Los Angeles) [6:40 PM] 
okay, so if you open `fec/static/js/home.js`


[6:40] 
you see `require('./vendor/tablist').init();`


[6:40] 
it used to just be used on the homepage, but now since we are using this pattern on the meetings page as well


[6:40] 
we need to include it globally in the js


[6:40] 
so move that line to `fec.js`

File: fec/home/templates/home/commission_meetings.html [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+      </div>
+    </section>
+  {% endblock %}
+ </div>
+</div>
+</section>
+{% endblock %}
+
+{% block extra_js %}
+  <script type="text/javascript" src="{% static 'js/fec.js' %}"></script>
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: You can remove this because `fec.js` is already included in the base template.

**Comment by johnnyporkchops** [comment]
aug-03-2017 13-14-01

File: fec/home/templates/home/commission_meetings.html [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+              </li>
+            </ul>
+          </nav>
+        </div>
+        <div class="main__content--right-full">
+          <section id="meetings-page-meetings-open-meetings" class="option" role="tabpanel" aria-hidden="true" aria-labelledby="open-meetings">
+            <h2 id="meetings-page-meetings-open-meetings">Open Meetings</h2>
+             <p>Open to the public. The Commission considers new regulations, advisory opinions and other public matters at open meetings, which are typically held at FEC headquarters on Thursdays at 10:00am. Commission meeting agendas are usually published a week before a scheduled meeting.</p>
+            <div class="filters--horizontal">
+             {% block filters1 %}
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: What are the filters blocks for?

File: fec/home/templates/home/commission_meetings.html [hunk 11]
```diff
@@ -0,0 +1,254 @@
...
+                    href="#meetings-page-meetings-executive-sessions"
+                    aria-selected="
+                    {% for key,value in request.GET.items %}{% ifequal value 'executive-sessions' %}true{% endifequal %}{% endfor %}
+                    {% for key,value in request.GET.items %}{% ifnotequal value 'executive-sessions' %}false{% endifnotequal %}{% endfor %}"
+                    >Executive Sessions</a>
+              </li>
+            </ul>
+          </nav>
+        </div>
+        <div class="main__content--right-full">
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: This can just be `main__content--right`

File: fec/home/templates/home/commission_meetings.html [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+          <section id="meetings-page-meetings-open-meetings" class="option" role="tabpanel" aria-hidden="true" aria-labelledby="open-meetings">
```
Comment (xtine) [inline fec/home/templates/home/commission_meetings.html]: Remove `class="option"` because it doesn't need the top border, per design.

**Comment by johnnyporkchops** [comment]
Apologies for all the commits to this, the code got a little messy when I had to revert to an older ref, just making sure its clean and up to date. It also included @jenniferthibault 's commits

File: fec/fec/templates/partials/meeting.html [hunk 13]
```diff
@@ -0,0 +1,17 @@
+{% load filters %}
+        <tr class="simple-table__row">
```
Comment (noahmanger) [inline fec/fec/templates/partials/meeting.html]: Remove extraneous indentation.

File: fec/home/templates/home/commission_meetings.html [hunk 14]
```diff
@@ -0,0 +1,237 @@
...
+      <div class="container">
+        <div class="sidebar-container sidebar-container--left">
+          <nav class="sidebar side-nav-alt">
+            <ul class="tablist" role="tablist" data-name="tab">
+              <li class="side-nav__item" role="presentation">
+                <a class="side-nav__link" id="openmeetings_tab"
+                    role="tab"
+                    data-name="open-meetings"
+                    tabindex="0"
+                    aria-controls="meetings-page-meetings-open-meetings"
```
Comment (noahmanger) [inline fec/home/templates/home/commission_meetings.html]: Small style thing: I'd just shorten this to `meetings`. As long as there's not a conflict, shorter is easier to read.

File: fec/home/templates/home/commission_meetings.html [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+</div>
+<section class="main__content--full data-container__wrapper" id="feed">
+  <div class="container">
+    <section id="meetings-page-meetings" class="tab-interface u-padding--bottom">
+      <div class="container">
+        <div class="sidebar-container sidebar-container--left">
+          <nav class="sidebar side-nav-alt">
+            <ul class="tablist" role="tablist" data-name="tab">
+              <li class="side-nav__item" role="presentation">
+                <a class="side-nav__link" id="openmeetings_tab"
```
Comment (noahmanger) [inline fec/home/templates/home/commission_meetings.html]: Small style thing: IDs should use `-` not `_`.

Comment (noahmanger) [inline fec/home/templates/home/commission_meetings.html]: And the `<a>` elements in the tabs don't need IDs.

File: fec/home/templates/home/commission_meetings.html [hunk 16]
```diff
@@ -0,0 +1,237 @@
...
+              </li>
+            </ul>
+          </nav>
+        </div>
+        <div class="main__content--right">
+          <section id="meetings-page-meetings-open-meetings" role="tabpanel" aria-hidden="true" aria-labelledby="open-meetings">
+            <h2 id="meetings-page-meetings-open-meetings">Open Meetings</h2>
+             <p>Open to the public. The Commission considers new regulations, advisory opinions and other public matters at open meetings, which are typically held at FEC headquarters on Thursdays at 10:00am. Commission meeting agendas are usually published a week before a scheduled meeting.</p>
+            <div class="filters--horizontal">
+             {% block filters1 %}
```
Comment (noahmanger) [inline fec/home/templates/home/commission_meetings.html]: What are these `{% block filters1 %}` and `{% block feed1 %}` for?

File: fec/home/templates/home/commission_meetings.html [hunk 17]
```diff
@@ -0,0 +1,237 @@
...
+                  <tbody>
+                    {% for open_meeting in open_meetings %}
+                      {% include 'partials/meeting.html' with meeting=open_meeting %}
+                     {% endfor %}
+                  </tbody>
+               </table>
+              <div class="results-info">
+                <span>Page {{ open_meetings.number }} of {{ open_meetings.paginator.num_pages }}</span>
+                  {% if open_meetings.has_previous %}
+                    <a class="button button--standard button--previous" href="?page={{ open_meetings.previous_page_number }}{% for key,value in request.GET.items %}{% ifnotequal key 'page' %}&amp;{{ key }}={{ value }}{% endifnotequal %}{% endfor %}"><span class="u-visually-hidden">Previous</span></a>
```
Comment (noahmanger) [inline fec/home/templates/home/commission_meetings.html]: Probably good to include the `?tab=` param in these next and previous links so you stay on the same tab you were on.

File: fec/home/views.py [hunk 18]
```diff
@@ -293,3 +298,155 @@ def serve_wagtail_doc(request, document_id, document_filename):
     """
     doc = get_object_or_404(Document, id=document_id)
     return HttpResponseRedirect(doc.file.url)
+
+def index_meetings(request):#,#  year=None, search=None, active=None):
+    meetings = MeetingPage.objects.live().order_by("-date")
+    open_meetings = meetings.filter(meeting_type ='O')
+    executive_sessions = meetings.filter(meeting_type ='E')
+    hearings= meetings.filter(title__contains='Hearing')
```
Comment (noahmanger) [inline fec/home/views.py]: Do we want to set a flag on the actual `MeetingPage` model for if it's a hearing or not? Or will it always have `hearing` in the title? Also, should these be excluded from the rest of the open meetings?

File: fec/home/views.py [hunk 19]
```diff
@@ -293,3 +298,155 @@ def serve_wagtail_doc(request, document_id, document_filename):
...
+        'years': years,
+
+        'years_h': years_h,
+        'hearings': hearings,
+
+        'years_e': years_e,
+        'executive_sessions': executive_sessions,
+        })
+
+def hearings(request):
```
Comment (noahmanger) [inline fec/home/views.py]: Why does the form need to submit to different URLs/views for each type if its just going to redirect back to `/meetings`?

**Comment by johnnyporkchops** [comment]
There is a problem with pagination--clicks on pagination arrows jump back to first tab. After trying a few different solutions things, the way I was able to get multiple pagination working on the same page was  use a unique page name for each pagination in the template like "page", "page_h" and "page_e". This all works except the template logic for the "Next >" and "< Previous" links (borrowed from updates view) uses an ifnotequal statement to render the href:
```
{% if hearings.has_next %}
                <a class="button button--standard button--next" href="?page_h={{ hearings.next_page_number }}{% for key,value in request.GET.items %}{% ifnotequal key 'page' %}&amp;{{ key }}={{ value }}{% endifnotequal %}{% endfor %}"><span class="u-visually-hidden">Next</span></a> {% endif %}
 ```
Basically It this statement says "if there is no key of page" then append the rest of the key-value pairs. 
So if one switches tabs and then tries to navigate pagination --since we have a key of page_e and page_h, it does not recognize them and the ifequal test writes nothing. So the query string becomes "http://127.0.0.1:8000/meetings/?page=e_2" which defaults to open meetings tab.I was trying to find a statement that would search the value for the existence of the string "page" within the a key so it would catch all page keys in the querystring, like :
```{% if not 'page' in key %}&amp;{{ key }}={{ value }}&amp;{{ key }}={{ value }}{% endif %} ```
But this does not seem to work.

**Comment by noahmanger** [comment]
@xtine can you give this a final look and approve your review if it looks good?

File: fec/home/views.py [hunk 20]
```diff
@@ -293,3 +296,92 @@ def serve_wagtail_doc(request, document_id, document_filename):
     """
     doc = get_object_or_404(Document, id=document_id)
     return HttpResponseRedirect(doc.file.url)
+
+def index_meetings(request):#,#  year=None, search=None, active=None):
```
Comment (xtine) [inline fec/home/views.py]: remove the comments

File: fec/fec/templates/partials/meeting.html [hunk 21]
```diff
@@ -0,0 +1,18 @@
+{% load filters %}
+  <tr class="simple-table__row">
+    <td class="simple-table__cell"><a href="{{ meeting.url }}">{% if hearing is meeting %} {{ meeting.title }}{% else %}{{ meeting.date|date:'F j, Y' }}{% endif %}</a></td>
+     <td class="simple-table__cell">
+      {% if meeting.sunshine_act_links %}
+        <ul>
+          {% for link in meeting.sunshine_act_links|splitlines %}
+          <li><a href="{{ link }}">{% if forloop.counter >= 2 %}Amended{% endif %}
+              Sunshine Act Notice
+              {% if meeting.sunshine_act_links|splitlines|length > 2 and forloop.counter >= 2 %}- {{ forloop.counter0 }}{% endif %}
```
Comment (xtine) [inline fec/fec/templates/partials/meeting.html]: Need to update the list items to match the design:
screen shot 2017-08-18 at 10 03 40 am
- bold date
- Sunshine Act list item separators

**Comment by noahmanger** [comment]
@xtine and @johnnyporkchops I went ahead and made those changes.
